Packs/iDefense/Integrations/iDefense_v2/iDefense_v2_test.py

```diff
@@ -0,0 +1,184 @@
+        {'total_size': 0, 'page': 1, 'page_size': 25, 'more': False},
+        "No results were found for ip 1.1.1.1"
+    )
+]
+
+
+@pytest.mark.parametrize('url, status_code, json_data, expected_output', MOCK_IP)
+def test_ip_command(url, status_code, json_data, expected_output):
+    """
+    Given:
+        - an IP
+
+    When:
+        - running ip command and validate whether the ip is malicious
+
+    Then:
+        - return command results containing indicator and dbotscore
+
+    """
+
+    ip_to_check = {'ip': '0.0.0.0'}
+    with requests_mock.Mocker() as m:
+        m.get(url, status_code=status_code, json=json_data)
+        client = Client(API_URL, 'api_token', True, False)
+        results = ip_command(client, ip_to_check)
+        output = results.to_context().get('EntryContext', {})
+        DBOT_KEY = 'DBotScore(val.Indicator && val.Indicator == obj.Indicator &&' \
+                   ' val.Vendor == obj.Vendor && val.Type == obj.Type)'
+        assert output.get('IP(val.Address && val.Address == obj.Address)', []) == expected_output.get('IP')
+        assert output.get(DBOT_KEY, []) == expected_output.get('DBOTSCORE')
+
+
+@pytest.mark.parametrize('url, status_code, json_data, expected_output', MOCK_WRONG_IP)
+def test_ip_not_found(url, status_code, json_data, expected_output):
+    """
+       Given:
+           - an IP
+
+       When:
+           - running ip command and validate whether the ip is malicious
+
+       Then:
+           - return command results with context indicate that no results were found
+
+       """
+    ip_to_check = {'ip': '1.1.1.1'}
+    with requests_mock.Mocker() as m:
+        m.get(url, status_code=status_code, json=json_data)
+        client = Client(API_URL, 'api_token', True, False)
+        results = ip_command(client, ip_to_check)
+        output = results.to_context().get('HumanReadable')
+        assert "No results were found for ip 1.1.1.1" in output
+
+
+def test_wrong_ip():
+    """
+       Given:
+           - an IP
+
+       When:
+           - running ip command validate first if the given ip is a valid ip
+
+       Then:
+           - raise error before calling http request that indicates that the given argument is not valid
+
+    """
+    ip_to_check = {'ip': '1'}
+    client = Client(API_URL, 'api_token', True, False)
+    try:
+        ip_command(client, ip_to_check)
+    except DemistoException as err:
+        assert "Invalid parameter was given" in str(err)
+
+
+def test_wrong_connection():
+    """
+          Given:
+              - an api token
+
+          When:
+              - checking api access
+
+          Then:
+              - raise error if there is no access because of wrong api token
+
+       """
+    from iDefense_v2 import test_module
+    with requests_mock.Mocker() as m:
+        mock_address = 'https://test.com/rest/threatindicator/v0/'
+        m.get(mock_address, status_code=401, json={})
+        client = Client(API_URL, 'wrong_token', True, False)
```

Comment:
for clarity purposes
```suggestion
        client = Client('bad_api_key', 'wrong_token', True, False)
```